hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/HRegionServer.java

```diff
@@ -789,8 +789,19 @@ public boolean registerService(com.google.protobuf.Service instance) {
     return true;
   }
 
-  private Configuration unsetClientZookeeperQuorum() {
+  private Configuration cleanupConfiguration() {
     Configuration conf = this.conf;
+    // We use ZKConnectionRegistry for all the internal communication, primarily for these reasons:
+    // - Decouples RS and master life cycles. For example, if all the masters are down, region
+    //   servers can abort at the same time, because the internal connection is master dependent and
+    //   fails.This is an operational nightmare. Using the ZK based registry means that the region
+    //   servers are now on the look out for new masters, if they are spun up.
```

Comment:
The tense here is confusing. 

"Using the ZK based registry means the region servers continue to be independent of master availability..." ?

Reply:
Clarified.